woocommerce/woocommerce-android
# [Coupons] Initial edit screen

Closes: #6358 

### Description
This PR adds the screen for editing the coupon, the functionality added so far is:
- Menu item for accessing the edit screen.
- Navigation.
- Fragment and ViewModel and Repository for the edit screen.
- Loading and starting a "coupon draft" from the stored coupon.
- Initial UI for the edit screen (the first part : "coupon details" is almost done UI wise).

### Testing instructions
1. Make sure the coupon beta switch is enabled.
2. Click on More menu.
3. Click on Coupons.
4. Click on one of the coupons.
5. Click on the menu overflow button.
6. Click on Edit coupon.
7. Make sure the navigation works, and the UI loads.
8. Check the screen title, the coupon amount and coupon code contains correct data (the rest is still not handled).

### Images/gif
<img width=400 src="https://user-images.githubusercontent.com/1657201/165876683-911f69cc-2b8d-4165-b3dc-a5ef6d8889d0.png"/>
